### DISCUSSION TITLE
Handle webhook callbacks with queue

### DISCUSSION THREAD

**Discussion by sjelfull** [craftcms/commerce#2368] [Ideas]
### Description

When Stripe fires 400+ webhooks when subscriptions renew, some of them will eventually fail because of the strain. Could you add a option to process the webhooks through the queue?

**Comment by philipboomy** [comment]
Is this still an issue?

**Comment by philipboomy** [comment]
I am gonna assume this is still an issue so will not use Commerce for 3000+ subscriptions.

**Comment by nfourtythree** [comment]
Hi @sjelfull & @philipboomy 

On the face of it, this sounds like an infrastructure issue more than a Commerce issue. This is similar to having 400 requests sent to `index.php`.

Processing a webhook has to happen in realtime because Stripe is expecting a response and that response has to have the correct data, we can't just reply `200 - ok` and process that data later in a queue in case there is an error. The webhook endpoint has to also deal with a multitude of different types of events where the response data is quite varied.

If you feel like there are certain events you believe can be handled asynchronously then you could select those specific events in Stripe and point them to your own custom controller action. You would then have free reign over handling things with a queue job.

Thanks.

**Comment by philipboomy** [comment]
@nfourtythree Appreciate you getting back to me on this. Helps me plan ahead.
